### ISSUE TITLE
Fututre-profing terminology for grouping (potential C++20 modules issue)

### ISSUE THREAD

**Issue by milasudril** [doxygen/doxygen#8772]
**Describe the bug**
Today, as has always been, Doxygen has its own internal notion of "modules". When modules are being introduced to C++, we have a name conflict.

**Expected behavior**
To be discussed within this thread

**Screenshots**
Not appliciable

**To Reproduce**
Not appiciable

**Version**
I guess target version for this issue is 2.0 (If Doxygen follows semver)

**Stack trace**
Not appliciable

**Additional context**

# Background

Doxygen already has the notion of "modules" (see https://www.doxygen.nl/manual/grouping.html). This may introduce a conflict when C++ modules are being deployed. One solution is to let C++ modules define a "module" in Doxygen as well. However, it may happen that you want to have a slightly different view within the documentation compared how the code is organized. Thus, I suggest that a module as read from the source code should not be the same thing as what a "module" in Doxygen represents today (the motivation is similar to why namespaces and modules are different things).  This raises the question what legacy modules should be called in the future.

Though adaption of C++ modules is likely to be slow, it is good to start a discussion about how to tread legacy modules before implementing support for C++ modules in Doxygen.

## Alt 1: Use `Categories`
From the documentation of the `\defgroup` command:

> Indicates that a comment block contains documentation for a group of classes, files or namespaces. This can be used to categorize classes, files or namespaces, and document those *categories*. You can also use groups as members of other groups, thus building a hierarchy of groups.

(emphasis mine)

This suggest that it could be better to rename legacy modules to categories.

## Alt 2: Use `Docset`

Since legacy modules can be used to organize the documentation it self, another choice could be to call them Docsets

**Comment by doxygen** [comment]
Then maybe just "group" is better. Rationale: the commands are already called `\defgroup` and `\ingroup`. The reason these groups ares shown as "modules" is because this was (and is) also what Qt does for their documentation, and I just copied that at the time (>20 years ago).

**Comment by milasudril** [comment]
Maybe an option to customize the section title, so you can set it to whatever fits the project.

**Comment by albert-github** [comment]
An aspect not mentioned, but that came up is the fact that the word module / modules is used on a different number of places and in a number of description. These are fortunately (and hopefully) all steered by means of translation functions:
```
    virtual QCString trModules()
    virtual QCString trModulesDescription()
    virtual QCString trModuleIndex()
    virtual QCString trModuleDocumentation()
    virtual QCString trModulesList()
    virtual QCString trModulesListDescription(bool extractAll)
    virtual QCString trModuleReference(const QCString &namespaceName)
    virtual QCString trModulesMembers()
    virtual QCString trModulesMemberDescription(bool extractAll)
    virtual QCString trModulesIndex()
    virtual QCString trModule(bool first_capital, bool singular)
```
and depending on the output language used these texts are slightly different (all depending on the context). 
A setting would be nice but I'm not sure it will be easy to catch all the different terms.
Also one has to be careful with the Fortran translations.

**Comment by milasudril** [comment]
Also, I realize that sometimes "module" is not a good synonym with a grouping concept. Here, "Events" does not form a module, but is more like an entity category:
 
![example](https://user-images.githubusercontent.com/734628/132987460-894a4b1a-1bd3-4f1d-9b0d-765d3e59456e.png)

In this example, it would be more appropriate to put "Events" one level up. Thus, what would be really good to have is a system where you can add stuff to the top level, in case a language does not support that particular feature. For C++-17 and older, you could have emulation of both "concepts" and "modules", For C, you can annotate interfaces. Some applications use messaging (I saw a request to support gpb). I thing there is a more general concept than a bunch of predefined sections.

**Comment by doxygen** [comment]
I have introduced C++20 module support with this [commit](https://github.com/doxygen/doxygen/commit/6d80fc7e5d03c259b1a7280972e0b28884217655). I renamed the existing "Modules" to "Topics".

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.8.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
